client/src/handlers/handlers.ts

```diff
@@ -14,6 +19,211 @@ export function makeHandleRunJob (
+function ensureNewFlixTerminal() {
+    const terminal = vscode.window.createTerminal(`flix-`+countTerminals.toString())
+    countTerminals+=1
+    return terminal
+}
+
+/**
+ * takes a string and a terminal and passes that string to the terminal.
+ *
+ * @param cmd string (a terminal command) to pass to the terminal.
+ *
+ * @param terminal vscode.Terminal 
+ *
+ * @return void
+*/
+function passCommandToTerminal(cmd:string, terminal: vscode.Terminal) {
+    terminal.show();
+    terminal.sendText(cmd);
+}
+
+/**
+ * Opens an input box to ask the user for input.
+ * uses the `vscode.window.showInputBox` function with custom `prompt` and `placeHolder` and value of `ignoreFocusOut` to be `true`.
+ *
+ *
+ * @return A promise that resolves to a string the user provided or to `undefined` in case of dismissal.
+*/
+async function takeInputFromUser() {
+    const input = await vscode.window.showInputBox({
+        prompt: "Enter arguments separated by spaces",
+        placeHolder: "arg0 arg1 arg2 ...",
+		ignoreFocusOut: true
+    })
+    return input
+}
+
+/**
+ * combines the paths of all flix files present in the current directory of vscode window.
+ * 
+ * gets an array of vscode.Uri for all flix files using `vscode.workspace.findFiles` function
+ *
+ * @return string of format "\<path_to_first_file\>" "\<path_to_second_file\>" ..........
+*/
+async function getFiles() {
+    const files = await vscode.workspace.findFiles(FLIX_GLOB_PATTERN)
+    let paths = files.map(x => x.path)
+    let cmd = quote(paths)
+    return cmd
+}
+
+/**
+ * sends a java command to compile and run flix program of vscode window to the terminal.
+ *
+ * @param terminal vscode.Terminal to receive the command. 
+ *
+ * @return void
+*/
+async function passArgs(terminal:vscode.Terminal, flixFilename: string) {
+
+    let cmd = "java -jar "+ flixFilename + " "
+    cmd += await getFiles()
+    let input = await takeInputFromUser()
+    if(!(input == undefined || input.trim().length == 0)) {
+        cmd += " --args ";
+        cmd += quote([input]);
+    }
+    passCommandToTerminal(cmd, terminal)  
+}
+
+/**
+ * Run main without any custom arguments
+ * 
+ * Sends command `java -jar <path_to_flix.jar> <paths_to_all_flix_files>` to an existing (if already exists else new) terminal.
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @return function handler
+*/
+export function cmdRunMain(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const globalStoragePath = context.globalStoragePath
+            const workspaceFolders = _.map(_.flow(_.get('uri'), _.get('fsPath')), vscode.workspace.workspaceFolders)
+            const flixFileLocation = await ensureFlixExists({ globalStoragePath, workspaceFolders, shouldUpdateFlix: launchOptions.shouldUpdateFlix })
+            var flixFilename = quote([flixFileLocation])
```

Comment:
See my comment above. 

Specifically I don't like that we call quote here, because it is easy to forget somewhere. Its better to always build a big array and then pass the whole thing to quote!